core/src/main/java/org/radargun/stages/stream/StreamStage.java

```diff
@@ -0,0 +1,214 @@
+package org.radargun.stages.stream;
+
+import org.radargun.DistStageAck;
+import org.radargun.StageResult;
+import org.radargun.config.Property;
+import org.radargun.config.Stage;
+import org.radargun.reporting.Report;
+import org.radargun.stages.AbstractDistStage;
+import org.radargun.stages.cache.RandomDataStage;
+import org.radargun.state.SlaveState;
+import org.radargun.stats.DataOperationStats;
+import org.radargun.stats.DefaultStatistics;
+import org.radargun.stats.Statistics;
+import org.radargun.traits.*;
+import org.radargun.utils.Projections;
+import org.radargun.utils.TimeService;
+import org.radargun.utils.Utils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Roman Macor &lt;rmacor@redhat.com&gt;
+ */
+@Stage(doc = "Stage which executes a specified stream Task against all keys in the cache.")
+public class StreamStage extends AbstractDistStage {
+
+   public static final String STREAM_RESULT_KEY = "streamResult";
+
+   @Property(optional = true, doc = "Boolean value that determines if the "
+           + "final results of the stream task are stored in the cache. "
+           + "The result object will be stored at key STREAM_RESULT_KEY. The default is false.")
+   private boolean storeResultInCache = false;
```

Comment:
So is this post-processed anywhere? StreamStage shouldn't be a direct copy. And it would be better if we could avoid having dependency on BasicOperations trait.